.buildkite/scripts/benchmark_master.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+echo "$BUILDKITE_PARALLEL_JOB"
+echo "$BUILDKITE_PARALLEL_JOB_COUNT"
+
+set -euox pipefail
+
+SYNTHETIC_SCRIPT="/workdir/examples/benchmark/synthetic_benchmark.py"
+
+function check_benchmark_log {
+    logfile=$1
+    speed=$3
+    loss=$4
+
+    final_batch_loss=$(cat ${logfile} | grep "TrainLoss" | tail -n 1 | awk '{print $4}')
+    img_per_sec=$(cat ${logfile} | grep "Img/sec per " | tail -n 1 | awk '{print $4}')
+
+    python -c "import sys; sys.exit(1) if float($final_batch_loss) != float($loss) else print('final_batch_loss is euqal.')"
+    python -c "import sys; sys.exit(1) if float($img_per_sec) < float($speed) else print('imag_per_sec is bigger than $speed.')"
+}
+
+export HOME=/workdir
+cd /workdir && pip install . && git clone https://github.com/BaguaSys/examples.git
+curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain stable -y
+source $HOME/.cargo/env
+pip install git+https://github.com/BaguaSys/bagua-core@master
+
+algorithms=(gradient_allreduce bytegrad decentralized)
```

Comment:
we also have qadam and low precision decentralized?

Reply:
done